editor/editor_file_system.cpp

```diff
@@ -2574,6 +2574,35 @@ void EditorFileSystem::move_group_file(const String &p_path, const String &p_new
 	}
 }
 
+void EditorFileSystem::add_new_directory(const String &p_path) {
```

Comment:
I'm wondering if this method could create the new directory on disk and update the FileSystem in memory. That was, the method could really be self contained and called directly from `EditorDirDialog`. That could simplify the process and `EditorDirDialog` is only used in `FileSystemDock` anyway.

The idea is that the EditorFileSystem should be the owner of the methods to execute files and folders manipulation in the project. The same concept could be implemented for: moving, renaming, copying, deleting files and folders, etc... The FileSystemDock would only have to call the EditorFileSystem.

Reply:
The directory is created by DirectoryCreateDialog before it emits `dir_created` signal.

Reply:
I understand and I think that most of the files operations are done in FileSystemDock already and mirrored in `EditorFileSystem` afterward.

Reply:
tbh now I think the EditorFileSystem could create the directory, but it should be called for EditorDirDialog (which makes the folder now), not FileSystemDock.